src/metabase/models/pulse.clj

```diff
@@ -0,0 +1,159 @@
+(ns metabase.models.pulse
+  (:require [clojure.tools.logging :as log]
+            [korma.core :as k]
+            [korma.db :as kdb]
+            [metabase.db :as db]
+            (metabase.models [card :refer [Card]]
+                             [common :refer [perms-readwrite]]
+                             [hydrate :refer :all]
+                             [interface :refer :all]
+                             [pulse-card :refer [PulseCard]]
+                             [pulse-channel :refer [PulseChannel] :as pulse-channel]
+                             [user :refer [User]])
+            [metabase.util :as u]))
+
+
+(defrecord PulseInstance []
+  ;; preserve normal IFn behavior so things like ((sel :one Database) :id) work correctly
+  clojure.lang.IFn
+  (invoke [this k]
+    (get this k)))
+
+(extend-ICanReadWrite PulseInstance :read :public-perms, :write :public-perms)
+
+
+(defentity Pulse
+  [(k/table :pulse)
+   (hydration-keys pulse)
+   timestamped]
+
+  (pre-insert [_ pulse]
+    (let [defaults {:public_perms perms-readwrite}]
+      (merge defaults pulse)))
+
+  (post-select [_ {:keys [id creator_id] :as pulse}]
+    (map->PulseInstance
+      (u/assoc* pulse
```

Comment:
I don't think you need to use `assoc*` here. That's just for doing multiple `assoc` calls at once that can reference the "intermediate" state of the map you're building using the symbol `<>`. (normal `assoc` adds all the new keys at once)

eg. instead of writing
```clojure
(let [field {}
      field (assoc field :table (delay (get-the-table-somehow field)))]
  (assoc field :db (delay (get-the-db-somehow @(:table field)))))
```

you can write:
```clojure
(assoc* {}
  :table (delay (get-the-table-somehow <>))
  :db    (delay (get-the-db-somehow @(:table <>))))
```

But if you're not using `<>` then you can just use regular `assoc`.

The name is probably a bit misleading, but it made sense to me when I wrote it way back in the stone age of "Clojurey corv". In other Lisp dialects, all bindings introduced by `let` are made "simultaneously" (they can't reference each other):

```emacs-lisp
(let ((x 100)
      (y (* x 2))) ; not ok! x not bound yet
  y)
```

But a variant called `let*` exists that works the same way as Clojure's `let`:


```emacs-lisp
(let* ((x 100)
       (y (* x 2))) ; ok
  y)
```

So then. Having to explain features of unrelated programming languages to explain the rational behind a function's name is probably a sign that the name is bad. I'll open a ticket to give it a better name so we can avoid this confusion in the future